Angle grinder AGAIN! Also, makes plasma cutters used for salvage instead of mining (and base gearpacks)

Thera-Pissed opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
re-implements #1716 as a base and adds cutter functionality to plasma cutters
also implements and uses gear packs from #2103 that never got in.
makes plasma cutters unable to mine, shorter range, and do burn instead of brute.
angle grinders are integrated into grinder packs, like water backtanks. Also draws power from a cell.
jackhammers can no longer break down walls and girders.
### Cutters can currently deconstruct:
- girders
- walls
- reinforced walls
- airlocks
- airlock frames
- grilles
- machine frames
- computer frames
- catwalks
- chairs
- beds
- tables
- lockers & crates
- salvage machines (the wasteplanet ones)
- railings
- lattice
- wooden barricades

cutters can also cut open safes and locked lockers & crates

attempted to revert map changes but using git checkout doesn't completely revert the gecko. Will try again later.

adds a mech salvage saw and a prototype energy saw. Doesn't make either available yet.

sprites by me
![image](https://github.com/shiptest-ss13/Shiptest/assets/90987989/65bd6b99-d63d-4c75-9227-a9987fddf9d2)

https://github.com/shiptest-ss13/Shiptest/assets/90987989/12262338-055f-4c7c-86d1-d31279ab953c

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Jackhammers as a main salvage tool is dumb, and angle grinders and cutters make more sense and have better functionality. Cutting apart old ruins and ships should be a reasonable source of material and shouldn't take years.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog



:cl:
add: angle grinders for salvage
add: reworks plasma cutters for salvage
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Commit:
adding angle grinder (no other object modified)

Commit:
make angle grinder work on some things and hurt eyes

Commit:
make the angle grinder work as a 80% circular saw for getho surgery

Commit:
typo in the last two commits

Commit:
deconstruct a bunch more stuff and allow to carry on back

Commit:
adding a proper icon to the grinde (sprite made by nirv)

Commit:
fixed deconstruct windows with grinder

Commit:
angle grinder at a satisfactory price ? (probaly temporary)

Commit:
added the /atom/proc/deconstruct_act(/mob/living/user, /obj/item/I) proc and the signal that goes with it

Commit:
typo

Commit:
salvage waste planet brocken machines

Commit:
force locked crates/closet

Commit:
make some feature toggleable at compilation

Commit:
config anglegrinder outpost availability

Commit:
angle grinder open safe, maybe ?

Commit:
linter

Commit:
i'm gonna eat the linter

Commit:
fix the angle grinder making light

Commit:
angle grinder now have it's sound yipee

Commit:
move config #defines and make them better

Commit:
deconstruct railing

Commit:
deconstruct machine FRAME

Commit:
correct a typo in the previous commit

Commit:
angle grinder deafen on use and you can now wear earmuff on neck

Commit:
linter ?

Commit:
guess who wote non compiling code to correct a linter error ?

Commit:
does the linter take blood tribute ?

Commit:
white space

Commit:
Apply the description changes suggested by zevotech

Commit:
deconstructing wooden baricade

Thera-Pissed committed:
gets code up to date and basic changes

Thera-Pissed committed:
more code updates

Thera-Pissed committed:
wield these

Martinpachu commented:
very rad

Thera-Pissed committed:
step 2. make it a backpack mount and take power

Thera-Pissed committed:
part 3 simple fixes

Thera-Pissed committed:
grilles and windows. windows get broken and are dumb to break

Thera-Pissed committed:
plastitanium

Thera-Pissed committed:
revives gear_packs by bjarl because theyre just better

Apogee-dev commented:
Can you remove breaching functionality from jackhammers, since there's now a specialty tool for that job? I never really liked the thing with jackhammers battering down metal walls anyway.

Martinpachu commented:
> Can you remove breaching functionality from jackhammers, since there's now a specialty tool for that job? I never really liked the

should replace them on the colossus and any other ships in that case

Martinpachu commented:
any other ships that use them for breaching i mean. i dont think any are using them for mining though.

Thera-Pissed commented:
I'll remove the jackhammer effect, though angle packs are cheap enough that I don't want to touch maps just yet. Which ships  would need their jackhammers replaced, though?

Martinpachu commented:
> I'll remove the jackhammer effect, though angle packs are cheap enough that I don't want to touch maps just yet. Which ships would need their jackhammers replaced, though?

colossus, hound (if you even care.....) gecko, hyena, inkwell and chronicle.

Thera-Pissed committed:
makes jackhammers not break walls good except concrete ones

Thera-Pissed committed:
adds grinders and normal cutters to colossus, gecko, and hyena

Thera-Pissed committed:
merge conflict

Thera-Pissed committed:
Merge branch 'master' into angle-grinder-again

Thera-Pissed committed:
actually updates hyena

Thera-Pissed commented:
couple maps tweaked to have roundstart cutters and packs

Thera-Pissed committed:
Merge branch 'master' into angle-grinder-again

Thera-Pissed committed:
I love .dmi merge conflicts

Thera-Pissed committed:
Merge branch 'master' into angle-grinder-again

Thera-Pissed committed:
merge conflict and preload cell fix

Thera-Pissed committed:
Merge branch 'angle-grinder-again' of https://github.com/Thera-Pissed/Shiptest into angle-grinder-again

JimKil3 commented:
the mudskipper has a surgical saw in a "salvage supplies" crate iirc which i'm pretty confident is meant to represent an angle grinder

Thera-Pissed committed:
conflict from temp fix

FalloutFalcon committed:
cleans up closest deconstuct logic

FalloutFalcon commented on code/modules/clothing/ears/_ears.dm:
```diff
@@ -15,6 +15,7 @@
 	name = "earmuffs"
 	desc = "Protects your hearing from loud noises, and quiet ones as well."
 	icon_state = "earmuffs"
+	slot_flags = ITEM_SLOT_EARS
```
pointless

FalloutFalcon commented on code/game/objects/structures/window.dm:
```diff
@@ -286,6 +287,15 @@
+	if(!I.tool_start_check(user, amount=0))
+		return FALSE
+	if (I.use_tool(src, user, decon_time, volume=100))
+		to_chat(user, "<span class='warning'>You shatter [src] with the [I].</span>")
```
```suggestion
		to_chat(user, span_warning("You shatter [src] with the [I]."))
```

FalloutFalcon commented on code/game/objects/structures/tables_racks.dm:
```diff
@@ -227,6 +227,15 @@
+	if(!I.tool_start_check(user, amount=0))
+		return FALSE
+	if (I.use_tool(src, user, 1 SECONDS, volume=0))
+		to_chat(user, "<span class='warning'>You cut [src] into sheets.</span>")
```
```suggestion
		to_chat(user, span_warning("You cut [src] into sheets."))
```

FalloutFalcon commented on code/game/objects/structures/girders.dm:
```diff
@@ -162,7 +143,7 @@
 						to_chat(user, "<span class='notice'>You fully reinforce the wall.</span>")
 						var/turf/T = get_turf(src)
-						T.PlaceOnTop(/turf/closed/wall/r_wall)
+						T.PlaceOnTop(/turf/closed/)
```
```suggestion
						T.PlaceOnTop(/turf/closed)
```

FalloutFalcon commented on code/modules/jobs/access.dm:
```diff
@@ -117,6 +117,8 @@
 		return TRUE
 
+	if(!item)
+		return FALSE
```
Already opened a seperate pr for this.

FalloutFalcon commented on code/modules/mining/equipment/angle_grinder.dm:
```diff
@@ -0,0 +1,93 @@
+	SIGNAL_HANDLER
+
+	playsound(src, 'sound/weapons/chainsawhit.ogg', 100, TRUE)
+	force = 24
```
mabye add a force_unwielded and force_wielded var for these

FalloutFalcon commented on code/modules/projectiles/guns/energy/special.dm:
```diff
@@ -127,10 +127,10 @@
 	usesound = list('sound/items/welder.ogg', 'sound/items/welder2.ogg')
-	tool_behaviour = TOOL_WELDER
-	toolspeed = 0.7 //plasmacutters can be used as welders, and are faster than standard welders
+	tool_behaviour = TOOL_DECONSTRUCT
```
should prob communicate with gristlebee about #3145..

Thera-Pissed replied:
communicated, going to wait until the PR is in to do discussed tweaks

FalloutFalcon reviewed (changes requested):
Good stuff, really support the plasma cutter changes
Where your checking for two separate tools is really asking for us to port tools having multiple tool qualities. Remind me at some point to work on that augh.
Could use some span defines on some stuff but I don't care all that much,

Thera-Pissed committed:
Merge pull request #2 from FalloutFalcon/quick-refactor-to-deconstuct

Quick refactor to deconstuct for closets

Thera-Pissed committed:
gear_handle instead of attachment

other PR uses obj/item/attachment and its getting in first. I will win the PR race one of these days.

Thera-Pissed committed:
Merge branch 'angle-grinder-again' of https://github.com/Thera-Pissed/Shiptest into angle-grinder-again

Thera-Pissed committed:
vsc + githubdesktop = head injury

Thera-Pissed committed:
Update angle_grinder.dm

Thera-Pissed committed:
adv cut effect, part of review, went down mech saw rabbit hole.

new green & purple slicing animation for advanced cutters (and when I add a funny angle grinder). Better to_chat from review. Mech saw that's currently the best saw available, but stops your mech until the task completes (otherwise you could psychic saw and walk away).

Thera-Pissed committed:
tweaks & energy saw & inhands

Thera-Pissed committed:
undoes map changes in prep for destructible walls

since destructible walls changes a load of maps, I'm waiting for that to go through before I tweak plasma cutters again.

Thera-Pissed commented:
waiting for #3145 to be merged before I continue working on this, since it overlaps a decent amount.

Thera-Pissed committed:
Merge remote-tracking branch 'upstream/master' into angle-grinder-again

Thera-Pissed committed:
destructable walls stuff

Thera-Pissed committed:
part of review

Thera-Pissed commented:
couple oddball things I want to work on but Im happy with the basic grinder and plasma cutter changes. not sure about plasma cutters in toolbelts from the destructible walls pr, but I'll wait and see how it goes.

FalloutFalcon commented on code/game/objects/structures/girders.dm:
```diff
@@ -162,7 +143,7 @@
 						to_chat(user, "<span class='notice'>You fully reinforce the wall.</span>")
 						var/turf/T = get_turf(src)
-						T.PlaceOnTop(/turf/closed/wall/r_wall)
+						T.PlaceOnTop(/turf/closed)
```
```suggestion
						T.PlaceOnTop(/turf/closed/wall/r_wall)
```

Thera-Pissed committed:
Update code/game/objects/structures/girders.dm

Co-authored-by: FalloutFalcon <86381784+FalloutFalcon@users.noreply.github.com>
Signed-off-by: HelmCrab <90987989+Thera-Pissed@users.noreply.github.com>

Thera-Pissed committed:
Merge remote-tracking branch 'upstream/master' into angle-grinder-again

Thera-Pissed committed:
Merge branch 'master' into angle-grinder-again

Thera-Pissed committed:
grinder works on walls again
